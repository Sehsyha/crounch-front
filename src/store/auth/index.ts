import { Module } from 'vuex';

import { TOKEN_STORAGE_KEY } from '@/utils/constants';
import { getters } from './getters';
import { actions } from './actions';
import { mutations } from './mutations';
import { RootState } from '..';

export interface AuthState {
  token: string;
  status: string;
}

export const state: AuthState = {
  token: localStorage.getItem(TOKEN_STORAGE_KEY) || '',
  status: '',
};

const namespaced: boolean = true;
export const authNamespace = { namespace: 'auth' };

export const auth: Module<AuthState, RootState> = {
  namespaced,
  state,
  getters,
  actions,
  mutations,
};
