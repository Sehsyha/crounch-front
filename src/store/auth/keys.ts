export enum AuthActions {
  LOGIN = 'LOGIN',
  SIGNUP = 'SIGNUP',
  LOGOUT = 'LOGOUT',
}

export enum AuthGetters {
  IS_AUTHENTICATED = 'IS_AUTHENTICATED',

}

export enum AuthMutations {
  LOGIN = 'LOGIN',
  LOGOUT = 'LOGOUT',
}
