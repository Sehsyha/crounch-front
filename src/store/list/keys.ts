export enum ListKeys {
  CREATE = 'CREATE',
  CREATED = 'CREATED',
  GETOWNERS = 'GETOWNERS',

  ADD = 'ADD',
  SET = 'SET',
  RESET = 'RESET',
};
