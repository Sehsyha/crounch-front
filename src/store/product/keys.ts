export enum ProductActions {
  CREATE = 'CREATE',
  CREATED = 'CREATED'
};

export enum ProductMutations {
  ADD = 'ADD',
  SET = 'SET',
  RESET = 'RESET',
};

export enum ProductGetters {
  GETALL = 'GETALL',
  GET = 'GET',
};
