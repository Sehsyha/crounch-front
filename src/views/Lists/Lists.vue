<template>
  <div>
    <h1>Listes</h1>
    <CreateList />
    <DisplayLists />
  </div>
</template>

<script lang="ts">
import Vue from 'vue'
import { Component } from 'vue-property-decorator';
import { Getter } from 'vuex-class';

import CreateList from '@/components/CreateList/CreateList.vue';
import DisplayLists from '@/components/DisplayLists/DisplayLists.vue';

@Component({
  components: {
    CreateList,
    DisplayLists,
  }
})
export default class Lists extends Vue { }
</script>
