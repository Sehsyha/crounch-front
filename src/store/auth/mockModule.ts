import { AuthKeys } from './keys';
import { Module } from 'vuex';
import { AuthState } from '.';
import { RootState } from '..';

const actions = {
  [AuthKeys.SIGNUP]: jest.fn(),
  [AuthKeys.LOGIN]: jest.fn(),
  [AuthKeys.LOGOUT]: jest.fn(),
};

const getters = {
  isAuthenticated: jest.fn(),
};

export function createAuthModuleMock(): Module<AuthState, RootState> {
  return {
    namespaced: true,
    actions,
    getters,
    state: {
      token: '',
      status: '',
    },
  };
}
