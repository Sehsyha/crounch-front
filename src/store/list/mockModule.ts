import { Module } from 'vuex';

import { ListKeys } from './keys';
import { ListState } from '.';
import { RootState } from '..';

const actions = {
  [ListKeys.CREATE]: jest.fn(),
  [ListKeys.CREATED]: jest.fn(),
  [ListKeys.GETOWNERS]: jest.fn(),
};

const getters = {
  lists: jest.fn(),
};

const mutations = {
  [ListKeys.RESET]: jest.fn(),
};

export function createListModuleMock(): Module<ListState, RootState> {
  return {
    namespaced: true,
    actions,
    getters,
    mutations,
    state: {
      lists: [],
    },
  };
}
