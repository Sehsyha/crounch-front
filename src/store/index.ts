import Vue from 'vue';
import Vuex, { StoreOptions } from 'vuex';
import { auth } from './auth';
import { ListModule } from './ListModule';
import { product } from './product'

Vue.use(Vuex);

export interface RootState { }

const store: StoreOptions<RootState> = {
  state: {},
  modules: {
    auth,
    ListModule,
    product,
  },
  strict: true,
};

export default new Vuex.Store<RootState>(store);
