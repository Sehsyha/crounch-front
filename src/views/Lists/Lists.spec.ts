import { shallowComponent } from '@/utils/test';
import Lists from './Lists.vue';
import { AuthState } from '@/store/auth';
import { RootState } from '@/store';
import { Module } from 'vuex';
import { Wrapper } from '@vue/test-utils';
import { createAuthModuleMock } from '@/store/auth/mockModule';
import { AuthGetters } from '@/store/auth/keys';

describe('Lists', () => {
  let wrapper: Wrapper<Lists>;
  let auth: Module<AuthState, RootState>;

  beforeEach(() => {
    auth = createAuthModuleMock();

    const modules = {
      auth,
    };

    wrapper = shallowComponent(Lists, { modules });
  });

  it('Should render.', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should render when user is authenticated.', () => {
    auth = createAuthModuleMock();
    (auth.getters![AuthGetters.IS_AUTHENTICATED] as jest.Mock).mockReturnValue(true);

    const modules = {
      auth,
    };

    wrapper = shallowComponent(Lists, { modules });

    expect(wrapper).toMatchSnapshot();
  });
});
