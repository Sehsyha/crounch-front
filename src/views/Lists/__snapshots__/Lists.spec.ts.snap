// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lists Should render. 1`] = `
<div>
  <h1>Listes</h1>
  <createlist-stub style="display: none;"></createlist-stub>
  <displaylists-stub></displaylists-stub>
</div>
`;
